mail/mailer.go

```diff
@@ -325,22 +330,20 @@ var recoverableErrorCodes = map[int]bool{
 // SendMail sends an email to the provided list of recipients. The email body
 // is simple text.
 func (m *MailerImpl) SendMail(to []string, subject, msg string) error {
-	m.stats.Inc("SendMail.Attempts", 1)
-
 	for {
 		err := m.sendOne(to, subject, msg)
 		if err == nil {
 			// If the error is nil, we sent the mail without issue. nice!
 			break
 		} else if err == io.EOF {
+			m.sendMailAttempts.WithLabelValues("failure, EOF").Inc()
```

Comment:
I think this should maybe be two labels? Something like: `result: failure`, `err: EOF` or `result:failure`, `err: smtp 421`. WDYT?

Reply:
:+1: